tests/timeseries/_battery_pool/test_battery_pool.py

```diff
@@ -130,27 +131,37 @@ async def setup_all_batteries(mocker: MockerFixture) -> AsyncIterator[SetupArgs]
     mock_microgrid = create_mock_microgrid(
         mocker, ComponentGraphConfig(batteries_num=2)
     )
-    sdk = SdkInterface(resampling_period_s=1)
+    min_update_interval: float = 0.2
+    # pylint: disable=protected-access
+    microgrid._data_pipeline._DATA_PIPELINE = None
+    microgrid._data_pipeline.initialize(ResamplerConfig(min_update_interval))
     streamer = MockComponentDataStreamer(mock_microgrid)
 
     # We don't use status channel from the sdk interface to limit
     # the scope of this tests. This tests should cover BatteryPool only.
     # We use our own battery status channel, where we easily control set of working
     # batteries.
-    battery_status_channel = Broadcast[BatteryStatus]("bat_status", resend_latest=True)
-    sender_channel = battery_status_channel.new_sender()
-    min_update_interval: float = 0.2
-    battery_pool = BatteryPool(
-        batteries_status_receiver=battery_status_channel.new_receiver(maxsize=1),
-        min_update_interval=timedelta(seconds=min_update_interval),
-    )
+    battery_pool = microgrid.battery_pool()
+
+    assert microgrid._data_pipeline._DATA_PIPELINE is not None
 
     args = SetupArgs(
-        battery_pool, min_update_interval, mock_microgrid, streamer, sender_channel
+        battery_pool,
+        min_update_interval,
+        mock_microgrid,
+        streamer,
+        microgrid._data_pipeline._DATA_PIPELINE._battery_status_channel.new_sender(),
     )
 
     yield args
-    await asyncio.gather(*[sdk.stop(), battery_pool.stop(), streamer.stop()])
+    await asyncio.gather(
+        *[
+            microgrid._data_pipeline._DATA_PIPELINE._stop(),
+            battery_pool.stop(),
+            streamer.stop(),
+        ]
+    )
+    # pylint: enable=protected-access
```

Comment:
yes,  that functions like a block,  you `disable` first,  then do all the pylint violations,  then `enable` back, so that the rest of the code isn't affected.